nsqd_connection.c

```diff
@@ -118,6 +119,12 @@ struct NSQDConnection *new_nsqd_connection(struct ev_loop *loop, const char *add
         NULL, NULL, nsqd_connection_error_cb,
         conn);
 
+    /* hack bufferrd socket */
+    free(conn->bs->read_buf);
+    free(conn->bs->write_buf);
+    conn->bs->read_buf = new_buffer(rdr->cfg->read_buf_len, rdr->cfg->read_buf_capacity);
+    conn->bs->write_buf = new_buffer(rdr->cfg->write_buf_len, rdr->cfg->write_buf_capacity);
```

Comment:
should we just make `new_buffered_socket` take arguments for initial size and capacity?